server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -89,19 +89,20 @@ default ActionListener<Response> delegateResponse(BiConsumer<ActionListener<Resp
     }
 
     /**
-     * Creates a listener that delegates all exceptions it receives to another listener.
+     * Creates a new listener, wrapping this one, that overrides {@link #onResponse(Object)} handling with the given {@code bc} lambda.
+     * {@link #onFailure(Exception)} handling is delegated to the wrapped internal listener. Exceptions in onResponse are not handled.
```

Comment:
I'm not sure about "internal" here, and "wrapped" sort of has other meanings in `ActionListener`-land too. Can we call it something else? "original listener" perhaps?

Also exceptions from `onResponse` are _forbidden_ (in tests at least).

Also (tiny nit) `onResponse` can be a `@link`

```suggestion
     * {@link #onFailure(Exception)} handling is delegated to the original listener. Exceptions in {@link #onResponse} are forbidden.
```